### PR TITLE
Update microwave

### PR DESCRIPTION
**Pull request by NoooneyDude**
- Creates a new microwave prefab that is a machine variant. The old microwave has been renamed to `BVOld_Microwave`.
- Prevent a possibility of hanging due to infinite loop when an invoked action at the end of a tool progress bar fails. (However, it seems the microwave deconstruction client hang is a different issue).
- Update MachineFrame.cs to use SpriteHandler's networked sprite updating, from using SyncVars.

Unfortunately, this does not fix the microwave hang issue on deconstruction.

### CODE REVIEW

**Comment by ThatDan123** [comment]
BoxStationV1: "Microwave" miss APC reference.
BoxStationV1: "Microwave" miss APC reference.
BoxStationV1: "Microwave" miss APC reference.
BoxStationV1: "Microwave" miss APC reference.
BoxStationV1: "Microwave" miss APC reference.
OutpostStation: "Microwave" miss APC reference.
OutpostStation: "Microwave" miss APC reference.
OutpostStation: "Microwave" miss APC reference.
OutpostStation: "Microwave" miss APC reference.
OutpostStation: "Microwave" miss APC reference.
FallStation: "Microwave" miss APC reference.
PogStation: "Microwave" miss APC reference.
PogStation: "Microwave" miss APC reference.
PogStation: "Microwave" miss APC reference.
PogStation: "Microwave" miss APC reference.
AncientTemple: "Microwave" miss APC reference.
Relicta: "Microwave" miss APC reference.
DarkJungle: "Microwave" miss APC reference.
DarkJungle: "Microwave" miss APC reference.
DarkJungle: "Microwave" miss APC reference.
LavaLand: "Microwave" miss APC reference.
LavaLand: "Microwave" miss APC reference.
SquareStation: "Microwave" miss APC reference.
SquareStation: "Microwave" miss APC reference.
SquareStation: "Microwave" miss APC reference.

BoxStationV1: "APC-EngineeringMaint" has a null value in the list.
BoxStationV1: "APC-Virology" has a null value in the list.
BoxStationV1: "APC-Kitchen" has a null value in the list.
BoxStationV1: "APC-Kitchen" has a null value in the list.
OutpostStation: "VIR_APC" has a null value in the list.
OutpostStation: "APC_Hallway_Dorms" has a null value in the list.
OutpostStation: "APC_Kitchen" has a null value in the list.
OutpostStation: "APC_Kitchen" has a null value in the list.
FallStation: "APC FOODCOURT" has a null value in the list.
PogStation: "APC - Food Service" has a null value in the list.
PogStation: "APC - Food Service" has a null value in the list.
PogStation: "Personell - APC" has a null value in the list.
AncientTemple: "APC-AT- General" has a null value in the list.
Relicta: "APC-RKitchen" has a null value in the list.
LavaLand: "APC Base1 - 3" has a null value in the list.
LavaLand: "APC Base1 - 3" has a null value in the list.
SquareStation: "APC - Kitchen" has a null value in the list.
SquareStation: "APC - Kitchen" has a null value in the list.

Missing reference found in scriptable object Microwave field Object
Missing reference found in scriptable object Microwave field Object
Missing reference found in scriptable object Microwave field Machine

File: UnityProject/Assets/Scripts/Construction/Machine.cs [hunk 1]
```diff
@@ -100,8 +100,14 @@ public void WhenDestroyed(DestructionInfo info)
 				itemStorage.ServerDropAll();
 			}
 
-			var frame = Spawn.ServerPrefab(framePrefab, SpawnDestination.At(gameObject)).GameObject;
-
+			SpawnResult frameSpawn = Spawn.ServerPrefab(framePrefab, SpawnDestination.At(gameObject));
+			if (!frameSpawn.Successful)
+			{
+				throw new MissingReferenceException(
```
Comment (corp-0) [inline UnityProject/Assets/Scripts/Construction/Machine.cs]: is it safe for us to throw exceptions? Maybe using the debug function for errors is prefered.

Comment (ThatDan123) [reply UnityProject/Assets/Scripts/Construction/Machine.cs]: Yeah better to do a 
`Debug.LogError("words");
return;
`

**Comment by corp-0** [comment]
Everything solved, I'm merging.

**Pull request merged**